src/main/java/chess/controller/ChessController.java

```diff
@@ -3,29 +3,56 @@
 import static chess.view.PositionConverter.convertToSourcePosition;
 import static chess.view.PositionConverter.convertToTargetPosition;
 
+import chess.dao.ChessBoardDao;
+import chess.dao.ChessGameDao;
 import chess.domain.ChessGame;
 import chess.domain.Position;
+import chess.domain.score.Score;
+import chess.domain.Team;
+import chess.domain.piece.Empty;
+import chess.domain.piece.Piece;
 import chess.view.GameCommand;
 import chess.view.InputView;
 import chess.view.OutputView;
+import java.util.Map;
+import java.util.Optional;
 
 public class ChessController {
 
     private final InputView inputView;
     private final OutputView outputView;
+    private final ChessGameDao chessGameDao;
+    private final ChessBoardDao chessBoardDao;
 
-    public ChessController(InputView inputView, OutputView outputView) {
+
+    public ChessController(InputView inputView, OutputView outputView,
+                           ChessGameDao chessGameDao, ChessBoardDao chessBoardDao) {
         this.inputView = inputView;
         this.outputView = outputView;
+        this.chessGameDao = chessGameDao;
+        this.chessBoardDao = chessBoardDao;
     }
 
     public void run() {
         outputView.printStartMessage();
         inputStartCommand();
-        ChessGame chessGame = ChessGame.createGame();
+        outputView.printGameList(chessGameDao.findAll());
+        String gameCommand = inputGameCommand();
+        ChessGame chessGame = startChessGame(gameCommand);
         progress(chessGame);
     }
 
+    private ChessGame startChessGame(String gameCommand) {
+        if (GameCommand.isNew(gameCommand)) {
+            long chessGameId = chessGameDao.create();
+            ChessGame chessGame = ChessGame.createGame(chessGameId);
+            chessBoardDao.save(chessGameId, chessGame.getBoard());
+            outputView.printGameId(chessGameId);
+            return chessGame;
+        }
+        return chessGameDao.findById(Integer.parseInt(gameCommand));
```

Comment:
이렇게 `if` 구문이 길어지는 경우엔 로직을 역전시켜보면 어떨까요?

```suggestion
        if (!GameCommand.isNew(gameCommand)) {
            return chessGameDao.findById(Integer.parseInt(gameCommand));
        }
        long chessGameId = chessGameDao.create();
        ChessGame chessGame = ChessGame.createGame(chessGameId);
        chessBoardDao.save(chessGameId, chessGame.getBoard());
        outputView.printGameId(chessGameId);
        return chessGame;
```